src/core/client/menus/vehicle.ts

```diff
@@ -9,56 +9,117 @@ import { IClientWheelItem, WheelMenu } from '../utility/wheelMenu';
+     * @memberof VehicleMenu
+     */
+    static addVehicleMenuInjections(
+        callback: (
+            player: alt.Player,
+            vehicle: alt.Vehicle,
+            options?: Array<IClientWheelItem>,
+        ) => Array<IClientWheelItem> | void,
+    ) {
+        vehicleMenuInjections.push(callback);
     }
 
-    const dist = distance(alt.Player.local.pos, vehicle.pos);
-    if (dist >= 4) {
-        return;
-    }
+    static openMenu() {
+        if (isAnyMenuOpen()) {
+            return;
+        }
+
+        const vehicle = getClosestVectorByPos<alt.Vehicle>(alt.Player.local.pos, alt.Vehicle.all, 'pos');
+        if (!vehicle || !vehicle.valid) {
+            return;
+        }
+
+        const dist = distance(alt.Player.local.pos, vehicle.pos);
+        if (dist >= 4) {
+            return;
+        }
+
+        let options: Array<IClientWheelItem> = [];
 
-    const options: Array<IClientWheelItem> = [];
+        if (!alt.Player.local.vehicle) {
+            const isDestroyed = native.getVehicleEngineHealth(vehicle.scriptID) <= 0;
 
-    if (!alt.Player.local.vehicle) {
-        const isDestroyed = native.getVehicleEngineHealth(vehicle.scriptID) <= 0;
+            // Not Pushing & Vehicle is Currently Unlocked
+            if (!PushVehicle.isPushing() && native.getVehicleDoorLockStatus(vehicle.scriptID) !== 2 && !isDestroyed) {
+                options.push({
+                    name: 'Push',
+                    callback: PushVehicle.start,
+                    data: [vehicle],
+                });
 
-        // Not Pushing & Vehicle is Currently Unlocked
-        if (!PushVehicle.isPushing() && native.getVehicleDoorLockStatus(vehicle.scriptID) !== 2 && !isDestroyed) {
-            options.push({
-                name: 'Push',
-                callback: PushVehicle.start,
-                data: [vehicle],
-            });
+                options.push({
+                    name: 'Open Storage',
+                    callback: () => {
+                        alt.emitServer(VEHICLE_EVENTS.OPEN_STORAGE, vehicle);
+                    },
+                });
+            } else {
+                options.push({
+                    name: 'Stop Push',
+                    callback: PushVehicle.clear,
+                });
+            }
 
-            options.push({
-                name: 'Open Storage',
-                callback: () => {
-                    alt.emitServer(VEHICLE_EVENTS.OPEN_STORAGE, vehicle);
-                },
-            });
-        } else {
-            options.push({
-                name: 'Stop Push',
-                callback: PushVehicle.clear
-            }); 
+            for (const callback of vehicleMenuInjections) {
+                try {
+                    const tempOptions = callback(alt.Player.local, vehicle, options);
+                    if (tempOptions) {
+                        options = tempOptions;
```

Comment:
My only issue with this is that the existing functionality of the vehicle should not be overwritten and should instead be appended to the existing array. Basically combining existing options and adding new options if that makes sense.

Additional wheel menus and options can be nested deeper inside of a wheel menu so I don't think a full replacement is the answer to this functionality.

Reply:
The replacement allows reordering and disabling some features like "push", which will not be interesting for all servers. Also, if a developer wants to only open storage if the door state of the trunk is open or to change the trunk door state while using the vehicle storage, it would be possible.

Reply:
Alright I can see the point. More flexibility and customization from core.